src/logic/basic.lean

```diff
@@ -501,6 +501,9 @@ end equality
 section quantifiers
 variables {α : Sort*} {β : Sort*} {p q : α → Prop} {b : Prop}
 
+lemma forall_imp (h : ∀ a, p a → q a) : (∀ a, p a) → ∀ a, q a :=
+λ h' a, h a (h' a)
```

Comment:
```suggestion
lemma forall_imp (h : ∀ a, p a → q a) (h' : ∀ a, p a) (a : α) : q a := h a (h' a)
```
Just some golfing.

Reply:
Will this obscure the point of the lemma at all or make it harder to find?  I'm not sure I would have noticed it in the second form.  I'll change it if you think it would be better, though.

Reply:
Lean should treat these exactly the same. I'm used to putting foralls before the colon, since then you can avoid the lambdas in the definition, but it doesn't really make a difference here.

Reply:
Could someone just verify that `library_search` finds the original statement, if Bryan's statement is what's available? I think it should, but it would be good to check here.

How about, if we go with the golfed version, we add a doc-string which contains @kmill's original statement, so the sake of findability?

Reply:
Oh, I'd also be happy with the original non-golfed version! I think I may have internalized too much of the "old" mathlib style...

Reply:
I weakly propose leaving it as-is just because it makes the sort of functoriality of `∀ a, ─` transparent.  If this lemma were in Lean core, it would be right before `forall_congr` in `library/init/logic.lean`, and this way both lemmas look the same except for the arrows. (This also explains why the parentheses are a little weird, though it looks like I forgot to copy the same parenthesization in `h`.)